autoPlay.user.js

```diff
@@ -325,6 +333,65 @@ function firstRun() {
 	enhanceTooltips();
 }
 
+function purchaseBadgeItems() {
+  // Spends badge points (BP's) when joining a new game.
+  // Dict contains the priority in terms of amount to buy (percentage of purchase). Probably a nicer way to do this...
+  // Priorities based on http://steamcommunity.com/groups/MSG2015/discussions/0/598198356167623323/
+  var abilityItemPriority = [
+    [ABILITIES.RAINING_GOLD, 20],
+    [ABILITIES.CRIT, 30],
+    [ABILITIES.TREASURE, 5],
+    [ABILITIES.WORMHOLE, 45]
+  ];
+
+  // Being extra paranoid about spending, since abilities update slowly.
+  var safeToBuy = true;
+  var intervalID = window.setInterval( function() {
+    var queueLen = s().m_rgPurchaseItemsQueue.length;
+    if (safeToBuy && queueLen > 0){
+      safeToBuy = false;
+    }
+    else if (!safeToBuy && queueLen === 0){
+      safeToBuy = true;
+    }
+  }, 100);
+
+  var buyItem = function(id) {
+    s().TrySpendBadgePoints(document.getElementById('purchase_abilityitem_' + id));
+  };
+
+  var badgePoints = s().m_rgPlayerTechTree.badge_points;
+
+  for (var i = 0; i < abilityItemPriority.length; i++) {
+    var abilityItem = abilityItemPriority[i];
+    var cost = w.$J(document.getElementById('purchase_abilityitem_' + abilityItem[0])).data('cost');
+
+    // Maximum amount to spend on each upgrade. i.e. 100 BP on item with a 10% share = 10 BP
+    var maxSpend = badgePoints * abilityItem[1] / 100;
+    var spent = 0;
+
+    // Don't over-spend the budget for each item, and don't overdraft on the BP
+    while (spent < maxSpend && cost <= s().m_rgPlayerTechTree.badge_points) {
+      if (!safeToBuy){
+        continue;
+      }
+      buyItem(abilityItem[0]);
+      spent += cost;
+    }
+  }
+
+  // Get any stragling 1 or 2 BP left over, using the last item (1 BP) in the priority array
+  while (s().m_rgPlayerTechTree.badge_points > 0) {
+    if (!safeToBuy) {
+      continue;
+    }
+    buyItem(abilityItemPriority[abilityItemPriority.length - 1][0]);
```

Comment:
The last item is the wormhole so it will not work in this case. A fallback should not use the priority list.

What you could is to buy the first item in the list (or the last enabled item until the badge points reachs 0.

Every item uses the css class `purchase_ability_item`.